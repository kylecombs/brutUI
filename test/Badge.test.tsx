import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Badge, images } from "../src";
import { FiArrowRight, FiStar } from "react-icons/fi";

afterEach(cleanup);

describe("Badge", () => {
  it("Renders badges correctly", () => {
    const { asFragment } = render(
      <>
        <Badge variant="gray" LeadingIcon={<FiStar />}>
          Label
        </Badge>

        <Badge variant="gray" size="sm" LeadingIcon={<FiStar />}>
          Label
        </Badge>

        <Badge
          variant="primary"
          size="md"
          LeadingIcon={<img src={images.NL} alt="nl" className="w-4 h-4" />}
        >
          Label
        </Badge>

        <Badge variant="error" size="lg" TrailingIcon={<FiArrowRight />}>
          Label
        </Badge>

        <Badge variant="warning" size="sm">
          Label
        </Badge>

        <Badge variant="error" size="sm">
          Label
        </Badge>
      </>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
