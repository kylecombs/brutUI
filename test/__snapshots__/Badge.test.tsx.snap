// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Renders badges correctly 1`] = `
<DocumentFragment>
  <div
    class="badge-base badge-gray badge-md "
  >
    <svg
      class="mr-1.5"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
      />
    </svg>
    Label
  </div>
  <div
    class="badge-base badge-gray badge-sm"
  >
    <svg
      class="mr-1.5"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
      />
    </svg>
    Label
  </div>
  <div
    class="badge-base badge-primary badge-md "
  >
    <img
      alt="nl"
      class="mr-1.5 w-4 h-4"
      src="https://res.cloudinary.com/tailwindcss/image/upload/v1635279277/nl_tpy2ab.svg"
    />
    Label
  </div>
  <div
    class="badge-base badge-error badge-lg"
  >
    Label
    <svg
      class="ml-1.5"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 5 19 12 12 19"
      />
    </svg>
  </div>
  <div
    class="badge-base badge-warning badge-sm"
  >
    Label
  </div>
  <div
    class="badge-base badge-error badge-sm"
  >
    Label
  </div>
</DocumentFragment>
`;
